drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -558,16 +668,21 @@ static int esp32_wifi_ap_enable(const struct device *dev,
 		return -EAGAIN;
 	}
 
-	net_eth_carrier_on(esp32_wifi_iface);
-
 	return 0;
 };
 
 static int esp32_wifi_ap_disable(const struct device *dev)
 {
 	int err = 0;
+	wifi_mode_t mode;
 
-	err = esp_wifi_stop();
+	esp_wifi_get_mode(&mode);
+	if (mode == ESP32_WIFI_MODE_APSTA) {
```

Comment:
Should you also check for `mode == ESP32_WIFI_MODE_AP`?

Reply:
We can if we want to, but in single iface mode, enabling `AP` and connecting to Wi-Fi will start the Wi-Fi in required mode.while disabling `AP` will stops it until enable AP or Connect to Wi-FI is requested. So, we can `mode == ESP32_WIFI_MODE_AP` check.